### ISSUE TITLE
[RFC] Our default parameters vs SMAC's default parameters. Can we do better?

### ISSUE THREAD

**Issue by MechCoder** [scikit-optimize/scikit-optimize#132]
Here is a table that compares our default parameters to SMAC parameters (http://arxiv.org/pdf/1211.0906v2.pdf). This will help in benchmarking | improving our defaults.

|  Parameters | SMAC | Our implementation |
| ------------- | ------------- | ---------- |
| n_estimators  | 10  | 100
| min_samples_leaf | | 3
| min_samples_split  | 5  |
| Variance | 0.01 | 0.00
| % features selected at node | 0.5 | all_features
| bootstrap | off | off

**Comment by MechCoder** [comment]
The acquisition function that they implemented is described in Eq 4 of https://www.cs.ubc.ca/~murphyk/Papers/gecco09.pdf .

**Comment by glouppe** [comment]
Also the solver for their acquisition function is a mix of random sampling over 10000 points and #109

**Comment by kegl** [comment]
Yes, I learned this second point from Frank at ICML. It seems crucial because it smoothes the forest. Alternatively, when sampling, IMHO care should be taken to randomly break ties.

**Comment by MechCoder** [comment]
I would be curious to know what are some good ways to break the ties?
It can be that our defaults are good enough but we are doing this part wrong.

**Comment by glouppe** [comment]
Not sure to follow. What ties are we talking about?

**Comment by MechCoder** [comment]
I was assuming the ties when new samples fall into the same leaf.

But thinking again, my interpretation might have been wrong since not sure how much a concern that is for a forest.

**Comment by kegl** [comment]
What I understood from Frank is that broad flat regions are a problem, which is why they are trying to smooth the forest. But I'm not sure. For now I would just break ties randomly. It's also a good principle to do sampling any time we can instead of taking the best point (eg in EI), but that's again just an intuition. I'm trying to understand why RF has this tendency to get stuck.

**Comment by glouppe** [comment]
Yes, in general trees are terrible at making predictions in regions outside of the training data: by construction those regions are assumed to be the same as the nearest points (which still may be far), with the same uncertainty. See e.g. figures from https://github.com/glouppe/notebooks/blob/master/Variance%20from%20tree-based%20models.ipynb The behaviour of GPs (cell 10) is good, because uncertainty increases outside of the training data, which will eventually make the exploration part to dominate. By contrast, in forests, the uncertainty is constant, which may blind the exploration of those regions (depending on the acquisition function). I think one of the most elegant and promising solution to this is Mondrian forests.

**Comment by kegl** [comment]
Eps-greedy? Just make every 10-20th points random.

**Comment by iaroslav-ai** [comment]
In [benchmarks](https://github.com/iaroslav-ai/scikit-optimize-benchmarks) that I run recently on [sigopt's evalset](https://github.com/sigopt/evalset) all`skopt`'s non - trivial algos seem to improve over SMAC3. All default setting were used for SMAC3, except that "deterministic": "true" was specified, which seems to be reasonable for the benchmarks used. So maybe default parameters of skopt are actually better than those of SMAC.  

This assumes that SMAC3 has the same defaults as in the referenced paper, which I will need to check tbh.
